src/erlcloud_sns.erl

```diff
@@ -276,7 +277,33 @@ list_endpoints_by_platform_application(PlatformApplicationArn, NextToken, Config
 list_endpoints_by_platform_application(PlatformApplicationArn, NextToken, AccessKeyID, SecretAccessKey) ->
     list_endpoints_by_platform_application(PlatformApplicationArn, NextToken, new_config(AccessKeyID, SecretAccessKey)).
 
+-spec list_topics/0 :: () -> [{topics, [[{arn, string()}]]} | {next_token, string()}].
+-spec list_topics/1 :: (undefined | string() | aws_config()) -> [{topics, [[{arn, string()}]]} | {next_token, string()}].
+-spec list_topics/2 :: (undefined | string(), aws_config()) ->  [{topics, [[{arn, string()}]]} | {next_token, string()}].
 
+list_topics() ->
+    list_topics(default_config()).
+
+list_topics(Config) when is_record(Config, aws_config) ->
```

Comment:
`when is_record(Config, aws_config)` is used everywhere in that file, so I repeated that pattern